iphone/Classes/TiUIListView.m

```diff
@@ -2041,74 +2030,52 @@ - (void)searchBarCancelButtonClicked:(UISearchBar *) searchBar
     [self buildResultsForSearchText];
 }
 
-#pragma mark - UISearchDisplayDelegate Methods
+#pragma mark - UISearchControllerDelegate
+
+- (void)willDismissSearchController:(UISearchController *)searchController
+{
+    self.searchString = @"";
+    [self buildResultsForSearchText];
+    if (_searchWrapper != nil) {
+        CGFloat rowWidth = floorf([self computeRowWidth:_tableView]);
+        if (rowWidth > 0) {
+            CGFloat right = _tableView.bounds.size.width - rowWidth;
+#ifndef TI_USE_AUTOLAYOUT
+            [_searchWrapper layoutProperties]->right = TiDimensionDip(right);
+#endif
+             [_searchWrapper refreshView:nil];
+        }
+    }
+    //IOS7 DP3. TableView seems to be adding the searchView to
+    //tableView. Bug on IOS7?
+    [self clearSearchController:self];
+    [_tableView reloadData];
```

Comment:
I guess this was copy-pasted over? So you have still been able to reproduce it without this hack?